lib/pages/create_post/create_post_url_field.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:lemmy_api_client/v3.dart';
+
+import '../../hooks/logged_in_action.dart';
+import '../../hooks/stores.dart';
+import '../../l10n/l10n.dart';
+import '../../util/files.dart';
+import '../../util/observer_consumers.dart';
+import 'create_post_store.dart';
+
+class CreatePostUrlField extends HookWidget {
+  final FocusNode titleFocusNode;
+
+  const CreatePostUrlField(this.titleFocusNode);
+
+  @override
+  Widget build(BuildContext context) {
+    final controller = useTextEditingController(
+      text: context.read<CreatePostStore>().url,
+    );
+    final loggedInAction = useLoggedInAction(
+      useStore((CreatePostStore store) => store.instanceHost),
+    );
+
+    uploadImage(Jwt token) async {
+      final pic = await pickImage();
+
+      // pic is null when the picker was cancelled
+      if (pic != null) {
+        await context.read<CreatePostStore>().uploadImage(pic.path, token);
+      }
+    }
+
+    return ObserverConsumer<CreatePostStore>(
+      listener: (context, store) {
+        // needed since flutter's TextFields cannot work as dumb widgets
```

Comment:
plz edit the comment to make it clear that it's this way so that picture uploading can work

Reply:
This is not true tho, without it picture uploading will work fine. This is only to show on the UI the newest URL